### PR TITLE
Set accept header in curl examples

### PR DESCRIPTION
**Pull request by therve**
### What does this PR do?

Add accept headers in curl examples.

### Motivation

Recent breakage in some APIs.

### Preview

Check curl examples in various sections
https://docs-staging.datadoghq.com/therve/curl-accept/api/latest/dashboards/#update-a-dashboard
https://docs-staging.datadoghq.com/therve/curl-accept/api/latest/dashboards/#delete-dashboards

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by davidejones** [comment]
Just noticed with these changes that the `\` is on a new line now instead of at the end of the first line e.g

```
# Curl command
curl -X GET "https://api.datadoghq.com/api/v1/dashboard"
 \
-H "Accept: "application/json" \
-H "DD-API-KEY: ${DD_API_KEY}" \
-H "DD-APPLICATION-KEY: ${DD_APP_KEY}"
```

**Comment by therve** [comment]
> Just noticed with these changes that the `\` is on a new line now instead of at the end of the first line e.g
> 
> ```
> # Curl command
> curl -X GET "https://api.datadoghq.com/api/v1/dashboard"
>  \
> -H "Accept: "application/json" \
> -H "DD-API-KEY: ${DD_API_KEY}" \
> -H "DD-APPLICATION-KEY: ${DD_APP_KEY}"
> ```

Yeah I don't know why... But I also want to make sure we only set it when the response can't be 204, need to spend more time investigating.

File: layouts/partials/api/curl.html [hunk 1]
```diff
@@ -53,11 +53,16 @@
         {{- end -}}
     {{- end -}}
 {{- end -}}
-<span class="kn">"</span>
-
+<span class="kn">"</span>&nbsp;\
+<span class="o">-H</span> <span class="s1">"Accept: "application/json"</span>
```
Comment (kayayarai) [inline layouts/partials/api/curl.html:57]:
```suggestion
<span class="o">-H</span> <span class="s1">"Accept: application/json"</span>
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
There's either a superfluous " or a missing "

**Review by bgdeutsch — APPROVED** [review]

**Review by buraizu — APPROVED** [review]

**Pull request merged**